TimeZoneが未対応だったら落ちちゃう

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{jp.lovesalmon.globalclock/jp.lovesalmon.globalclock.activity.TabActivity}: org.threeten.bp.zone.ZoneRulesException: Unknown time-zone ID: JST
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
	at android.app.ActivityThread.-wrap12(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6119)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
Caused by: org.threeten.bp.zone.ZoneRulesException: Unknown time-zone ID: JST
	at org.threeten.bp.zone.ZoneRulesProvider.getProvider(ZoneRulesProvider.java:178)
	at org.threeten.bp.zone.ZoneRulesProvider.getRules(ZoneRulesProvider.java:133)
	at org.threeten.bp.ZoneRegion.ofId(ZoneRegion.java:143)
	at org.threeten.bp.ZoneId.of(ZoneId.java:357)
	at jp.lovesalmon.globalclock.model.CompareLocale.<init>(CompareLocale.java:56)
	at jp.lovesalmon.globalclock.activity.TabActivity.setDefaultLocale(TabActivity.java:213)
	at jp.lovesalmon.globalclock.activity.TabActivity.onCreate(TabActivity.java:61)
	at android.app.Activity.performCreate(Activity.java:6679)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2618)
	... 9 more
```

--- a/app/src/main/java/jp/lovesalmon/globalclock/activity/TabActivity.java
+++ b/app/src/main/java/jp/lovesalmon/globalclock/activity/TabActivity.java
@@ -210,7 +210,7 @@ private final static String PREF = "pref";
             String id = (String)map.get("id");
 
             if (defaultIds.contains(id)) {
-                CompareLocale locale = new CompareLocale(this);
+                CompareLocale locale = CompareLocale.getInstance();
                 locale.setLocationName(name);
                 locale.setGmtId(id);
                 ZonedDateTime newOne = locale.getZonedDateTime().withZoneSameInstant(ZoneId.of(gmt));

--- a/app/src/main/java/jp/lovesalmon/globalclock/fragment/SearchListFragment.java
+++ b/app/src/main/java/jp/lovesalmon/globalclock/fragment/SearchListFragment.java
@@ -167,7 +167,7 @@ public class SearchListFragment extends Fragment {
         public void onItemClick(AdapterView<?> adapterView, View view, int position, long id) {
             // set locale.
             TmpTimeZone tmpTimeZone = getItem(position);
-            CompareLocale locale = new CompareLocale(getActivity());
+            CompareLocale locale = CompareLocale.getInstance();
             locale.setLocationName(tmpTimeZone.getName());
             locale.setGmtId(tmpTimeZone.getLocaleId());
             ZonedDateTime newOne = locale.getZonedDateTime().withZoneSameInstant(ZoneId.of(tmpTimeZone.getGmt()));

--- a/app/src/main/java/jp/lovesalmon/globalclock/model/CompareLocale.java
+++ b/app/src/main/java/jp/lovesalmon/globalclock/model/CompareLocale.java
@@ -1,6 +1,5 @@
 package jp.lovesalmon.globalclock.model;
 
-import android.content.Context;
 import android.provider.BaseColumns;
 import android.support.annotation.NonNull;
 
@@ -12,10 +11,6 @@ import com.github.gfx.android.orma.annotation.Table;
 
 import org.threeten.bp.ZoneId;
 import org.threeten.bp.ZonedDateTime;
-
-import java.util.TimeZone;
-
-import jp.lovesalmon.globalclock.utils.AdapterGenerater;
 
 @Table
 public class CompareLocale implements ListBindingInterface {
@@ -46,18 +41,15 @@ public class CompareLocale implements ListBindingInterface {
 
     private int offsetMins = 0;
 
-    public CompareLocale() {
-
-    }
-
-    public CompareLocale(Context context) {
-        this.basis = false;
-        String gmt = TimeZone.getDefault().getDisplayName(false, TimeZone.SHORT);
-        this.zonedDateTime = ZonedDateTime.now(ZoneId.of(gmt))
-                    .withSecond(0).withNano(0);
-        this.gmtId = AdapterGenerater.getIdByGMT(context, gmt);
-        this.taskName = "";
-        this.locationName = gmtId;
+    public static CompareLocale getInstance() {
+        CompareLocale newInstance = new CompareLocale();
+        newInstance.basis = false;
+        newInstance.zonedDateTime = ZonedDateTime.now(ZoneId.systemDefault())
+                .withSecond(0).withNano(0);
+        newInstance.gmtId = ZoneId.systemDefault().getId();
+        newInstance.taskName = "";
+        newInstance.locationName = newInstance.gmtId;
+        return newInstance;
     }
 
     @Getter

